site/src/sphinx/client-circuit-breaker.rst

```diff
@@ -1,6 +1,270 @@
+         +---+                             |        |                        |
+         |   |                             |        |                        |
+         |   v                             |        |                        |
+    +----+-----------+  exceed threshold   |        |     timeout         +--+-------------+
+    |                +---------------------+        +-------------------->|                |
+    |     CLOSED     |                                                    |    HALF_OPEN   |
+    |                |<---------------------------------------------------+                |
+    +----------------+            back to normal(request succeed)         +----------------+
+
+    @endditaa
+
+``CircuitBreakerClient``
+------------------------
+
+Armeria provides two different :api:`Client` implementations depending on the
+:api:`Request` and :api:`Response` types:
+
+- :api:`CircuitBreakerHttpClient`
+- :api:`CircuitBreakerRpcClient`
+
+Let's use :api:`CircuitBreakerHttpClient` to find out what we can do.
+You can use the decorator_ method in :api:`ClientBuilder` to build a :api:`CircuitBreakerHttpClient`:
+
+.. code-block:: java
+
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreaker;
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerHttpClient;
+    import com.linecorp.armeria.client.circuitbreaker.CircuitBreakerStrategy;
+    import com.linecorp.armeria.client.ClientBuilder;
+    import com.linecorp.armeria.client.HttpClient;
+    import com.linecorp.armeria.common.HttpRequest;
+    import com.linecorp.armeria.common.HttpResponse;
+
+    final CircuitBreakerStrategy<HttpResponse> strategy = CircuitBreakerStrategy.onServerErrorStatus();
+    final CircuitBreaker circuitBreaker = CircuitBreaker.ofDefaultName();
+    final HttpClient client = new ClientBuilder(...)
+            .decorator(HttpRequest.class, HttpResponse.class,
+                       CircuitBreakerHttpClient.newDecorator(circuitBreaker, strategy))
+            .build(HttpClient.class);
+
+    client.execute(...).aggregate().join(); // Send requests on and on.
+
+Now, the :api:`Client` can track the number of success or failure events depending on the :apiplural:`Response`.
+The :api:`CircuitBreaker` will be ``OPEN``, when the number of failures divided by the total number of
+:apiplural:`Request` exceeds the failure rate. Then the :api:`Client` will immediately get
+:api:`FailFastException` by the :api:`CircuitBreaker`.
+
+.. _circuit-breaker-strategy:
+
+``CircuitBreakerStrategy``
+--------------------------
+
+How does the :api:`CircuitBreaker` know whether a :api:`Response` is successful or not?
+:api:`CircuitBreakerStrategy` does the job. In the above example, if a :api:`Response`'s status is ``5xx``
+or an ``Exception`` is raised during the call, the count of failure is increased.
+Of course, you can have your own ``strategy`` by implementing :api:`CircuitBreakerStrategy`.
+
```

Comment:
Could you add some explanation about the example below? e.g. The following example shows a CircuitBreakerStrategy implementation that ...: